Grandparent Inbox and PostView UI

RebeccaLyric opened the pull request:
- Limits the posts fetched for a grandparent to 6. Before we have pagination this keeps the grandparent view in order.

- Updated Inbox UI: message icon and coloration for read/unread
![Screen Shot 2020-05-06 at 11 50 01 AM](https://user-images.githubusercontent.com/34624717/81205090-d2ec0580-8f8f-11ea-9a8e-00874ea0a1de.png)

- Updated grandparent post view: Should display all photo and/or text content within the box. More testing is definitely welcome.
![Screen Shot 2020-05-06 at 11 52 31 AM](https://user-images.githubusercontent.com/34624717/81205278-16467400-8f90-11ea-9cd5-d67aabb46355.png)
![Screen Shot 2020-05-06 at 11 53 02 AM](https://user-images.githubusercontent.com/34624717/81205342-28281700-8f90-11ea-8b82-b443d2ecd774.png)
![Screen Shot 2020-05-06 at 11 58 58 AM](https://user-images.githubusercontent.com/34624717/81205939-fcf1f780-8f90-11ea-8e98-197a567cbd1b.png)

To keep things manageable, this PR does not include UI adjustments for the second half of the grandparent flow, i.e., replying to a post. There's a lot going on already 😅

woodkri commented:
Overall this looks great! The cutoff looks perfect for the post I sent with both a photo and long text, but it ran over with just long text, maybe due to different screen size or different word lengths? Screenshot attached.

I have a question on the cutoff text. Is there a place the grandparents can view the full text, or should I be imposing character limits on the family side (if so, what works best?)?

I also noticed the PR is into master. If you want to change it to week6-master, you can edit the PR and change the branch to merge into. I did the same thing yesterday!

![grandparentSideLongText](https://user-images.githubusercontent.com/33300923/81235537-2af32e00-8fc9-11ea-95c2-932e617b5507.PNG)

RebeccaLyric commented:
Thanks for your comments and testing the text @woodkri !

I changed the PR to merge into week6-master. Thanks for the reminder.

A modal to view the full text / full-size photo is pending. Part of the reason I haven't done that yet is I did want to discuss whether we wanted to have that additional click for the grandparent? If there isn't a separate modal then a character limit would be necessary. However it would be different for posts with a photo versus text only. Let's talk about that. 

But yeah that's interesting that the text overflowed 🤔 The Box size is fixed to a specific height and width so the same number of characters should fit regardless. I set the limit using your function that I copy/pasted into post.ts and added a character-limit parameter. I will definitely look into that.

RebeccaLyric commented:
I updated the styling to: 

1. Add a magnifying button that when clicked will allow the grandparent to open a modal and see the photo larger. Note: The modal does not exist yet but clicking on the magnifying glass or on the photo should log in the console that this is where the modal will open.

2. Accommodate different photo sizes and text lengths. It's not perfect but it looks pretty decent on most of the combinations I tested. For now the text of messages is truncated but I will work with @woodkri to set a limit for the poster.
